packages/network-enablement-controller/src/utils.ts

```diff
@@ -0,0 +1,88 @@
+import { toHex } from '@metamask/controller-utils';
+import { toEvmCaipChainId } from '@metamask/multichain-network-controller';
+import type { CaipChainId, CaipNamespace, Hex } from '@metamask/utils';
+import {
+  isCaipChainId,
+  isHexString,
+  KnownCaipNamespace,
+  parseCaipChainId,
+} from '@metamask/utils';
+
+import type { NetworkEnablementControllerState } from './NetworkEnablementController';
+
+/**
+ * Derives the namespace, storage key, and CAIP chain ID from a given chain ID.
+ *
+ * This utility function handles the conversion between different chain ID formats.
+ * For EVM networks, it converts Hex chain IDs to CAIP-2 format and determines
+ * the appropriate storage key. For non-EVM networks, it parses the CAIP-2 chain ID
+ * and uses the full chain ID as the storage key.
+ *
+ * @param chainId - The chain ID to derive keys from (Hex or CAIP-2 format)
+ * @returns An object containing namespace, storageKey, and caipId
+ * @throws Error if the chain ID cannot be parsed
+ */
+export function deriveKeys(chainId: Hex | CaipChainId) {
+  const caipId: CaipChainId = isCaipChainId(chainId)
+    ? chainId
+    : toEvmCaipChainId(chainId);
+  const { namespace, reference } = parseCaipChainId(caipId);
+  let storageKey: CaipChainId | Hex;
+  if (namespace === (KnownCaipNamespace.Eip155 as string)) {
+    storageKey = isHexString(chainId) ? chainId : toHex(reference);
+  } else {
+    storageKey = caipId;
+  }
+  return { namespace, storageKey, caipId, reference };
+}
+
+/**
+ * Checks if the specified network is the only enabled network in its namespace.
+ *
+ * This function is used to prevent unnecessary state updates when trying to enable
+ * This method is used to prevent the last network in a namespace from being removed.
+ *
+ * @param state - The current controller state
+ * @param namespace - The namespace to check
+ * @param chainIdToCheck - The chain ID to check if it's the only enabled network
+ * @returns True if the network is the only enabled network in the namespace, false otherwise
+ */
+export function isOnlyNetworkEnabledInNamespace(
```

Comment:
When we call this function we've already parsed the given chain ID, so can we call this function with the `namespace`, `storageKey`, and `caipId` instead of the chain ID so we don't have to re-parse it below? (We could also assign a type to the return value of `deriveKeys` so we can just pass the entire object, whichever is easier.)